### PR TITLE
feat(payments): Add processing screen for PayPal invalid billing agre…

### PR DESCRIPTION
**Pull request by biancadanforth**
…ement modal

## Because

- If a subscriber cancels their billing agreement in PayPal, they have to reauthorize recurring payments to keep their subscription active.
- This requires [creating a new billing agreement and some other processing on the back-end](https://github.com/mozilla/fxa/blob/38834e96445ce64a33afcdd8738a96e65fe792d7/packages/fxa-auth-server/lib/routes/subscriptions/paypal.ts#L359-L378), so we should indicate to the user that this processing is taking place and whether it completed successfully or not.

## This pull request

- Adds a processing screen after the user successfully reauthorizes payment in PayPal for a subscription in the `PaymentUpdateForm` component of the Subscription Management page.
- Refreshes the page when processing is complete.
- Updates the `AlertBar` error message for funding source problems, since that error won't clear until the next time we run the [PayPal Processor script](https://github.com/mozilla/fxa/blob/38834e96445ce64a33afcdd8738a96e65fe792d7/packages/fxa-auth-server/scripts/paypal-processor.ts) (currently 1x/hour) and fixes a comma splice and incorrect localization ID for the original message.
- Makes the font size of the "Change" `ActionButton` the same size as the other buttons on the page.

## Issue that this pull request solves

Closes: #7945 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Screenshot_2021-03-31 Firefox Accounts
 
For the funding source error alert message, I added an extra sentence to inform the subscriber that the error may not clear immediately after they fix the problem in PayPal:

Before:
> Invalid payment information, there is an error with your account.

After:
> Invalid payment information; there is an error with your account. This alert may take some time to clear after you successfully update your information.

@jess-cook03: Does this message seem acceptable to you? Note: I _thought_ #7903 was going to handle the overlap onto the header (CC @chenba ), but that issue is closed, so it may need to be a separate follow-up here.

### CODE REVIEW

File: packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx [hunk 1]
```diff
@@ -101,18 +104,47 @@ export const PaymentUpdateForm = ({
...
+        Invalid payment information; there is an error with your account. This
+        alert may take some time to clear after you successfully update your
+        information. {actionButton}
+      </span>
+    </Localized>
+  );
+
+  const getPaypalErrorAlertBarContent = () => {
+    switch (paypal_payment_error) {
+      case 'missing_agreement': {
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx:134]: Ok, a suggestion.  I don't think it should block the merge, and I know you didn't originally introduce the type for `paypal_payment_error` for the frontend.

1. There's actually a type for it: https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts#L47  So we could move that to fxa-shared/subscriptions and use it as the type on the frontend as well.

2. Going further, we could additionally declare these values once somewhere and not use string literals for the cases here.

File: packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx [hunk 2]
```diff
@@ -177,13 +209,17 @@ export const PaymentUpdateForm = ({
           </AlertBar>
         )}
 
-        {paypal_payment_error && (
+        {!transactionInProgress && paypal_payment_error && (
           <AlertBar className="alert alertError">
             {errorAlertBarContent}
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx:214]: Inline `getPaypalErrorAlertBarContent()` here so that it is not called until the other two conditions are true?

File: packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx [hunk 3]
```diff
@@ -66,12 +67,14 @@ export const PaymentUpdateForm = ({
   const [submitNonce, refreshSubmitNonce] = useNonce();
   const [updateRevealed, revealUpdate, hideUpdate] = useBooleanState();
   const [inProgress, setInProgress, resetInProgress] = useBooleanState(false);
+  const [transactionInProgress, setTransactionInProgress] = useState(false);
```
Comment (chenba) [inline packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx:70]: Rename the `inProgress` vars on line 69 to reduce confusion?

**Review by chenba — APPROVED** [review]
👍

File: packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx (hunk 1 above)
Comment (biancadanforth) [reply packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx:134]: I can do 1) easily. Do you have a suggestion for where to put these values for 2)? I could put it in `fxa-shared/subscriptions/metadata.ts`... Is the idea something like:
```ts
export const PAYPAL_PAYMENT_ERROR_MISSING_AGREEMENT: PaypalPaymentError = 'missing_agreement';
export const PAYPAL_PAYMENT_ERROR_FUNDING_SOURCE: PaypalPaymentError = 'funding_source';
```
... and then import these consts where we use string literals?

File: packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx (hunk 2 above)
Comment (biancadanforth) [reply packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx:214]: That function doesn't create the React components unless there's a `paypal_payment_error`, but I could also avoid calling it entirely instead of every render, so I'll change it.

File: packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx (hunk 1 above)
Comment (chenba) [reply packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx]: For 2) we could create a `fxa-shared/subscriptions/constants.ts`, or, stick them in the same `types.ts` since I was thinking of defining the type with them.


```
export const PAYPAL_PAYMENT_ERROR_MISSING_AGREEMENT = 'missing_agreement';
export const PAYPAL_PAYMENT_ERROR_FUNDING_SOURCE = 'funding_source';

export type PaypalPaymentError = typeof PAYPAL_PAYMENT_ERROR_MISSING_AGREEMENT | typeof PAYPAL_PAYMENT_ERROR_FUNDING_SOURCE;
```

Or, you can use an object:

```
export const PaypalPaymentErrors = {
  missingAgreement: 'missing_agreement',
  fundingSoruce: 'funding_source',
} as const;
export type PaypalPaymentError = typeof PaypalPaymentErrors[keyof typeof PaypalPaymentErrors];
```

I'll leave the style choice to you.

**Comment by biancadanforth** [comment]
@chenba Here's the diff for the negative scenario as we discussed: https://github.com/mozilla/fxa/compare/337051f..48ced06 . The two test cases in `PaymentUpdateForm.test.tsx` are identical except for in one case, `apiUpdateBillingAgreement` succeeds and in the other, it fails.

File: packages/fxa-payments-server/public/locales/en-US/main.ftl [hunk 4]
```diff
@@ -309,7 +309,8 @@ sub-customer-error =
 sub-billing-update-success = Your billing information has been updated successfully
 sub-route-payment-modal-heading = Invalid billing information
 sub-route-payment-modal-copy = There seems to be an error with your Paypal account, we need you to take the necessary steps to resolve this payment issue.
```
Comment (chenba) [inline packages/fxa-payments-server/public/locales/en-US/main.ftl:311]: Request!  Let's replace 'Paypal' with `{ -brand-name-paypal }` on line 311 so we don't need to do it in another PR. 😁

Comment (biancadanforth) [reply packages/fxa-payments-server/public/locales/en-US/main.ftl:311]: Good call; I copied that from before, but you're right, we're not using the placeable, and also it's not "PayPal" (two capital Ps)
